File: 1b.cc
#include <fstream>
#include <iostream>

int main(void) {
    std::string line;
    std::ifstream fin;
    int floor = 0;
    int pos = 1;

    fin.open("1.dat");
    std::getline(fin, line);
    for (auto c : line) {
        if (c == '(')
            floor++;
        if (c == ')')
            floor--;
        if (floor == -1) {
            std::cout << pos << std::endl;
            break;
        }
        pos++;
    }
}

File: 1b.py
#!/usr/bin/python3

with open('1.dat') as f:
    data = f.read()
floor = 0
pos = 1
for char in data:
    if char == '(':
        floor += 1
    if char == ')':
        floor -= 1
    if floor == -1:
        print(pos)
        break
    pos += 1
